### PR TITLE
Feature/enable scaffolder dry run

### PR DESCRIPTION
**Pull request by rewixe**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend-module-github/src/actions/github.test.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ import {
   GithubCredentialsProvider,
   ScmIntegrations,
 } from '@backstage/integration';
+import { when } from 'jest-when';
```
Comment (benjdlambert) [inline plugins/scaffolder-backend-module-github/src/actions/github.test.ts:44]: Think you can remove this, we dropped this library from the scaffolder tests recently.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Hey :wave:

Thanks for this just a quick message about the `jest-when` library. Maybe comes from a bad rebase, but it's not needed anymore. 🙏

File: plugins/scaffolder-backend-module-github/src/actions/github.test.ts (hunk 1 above)
Comment (rewixe) [reply plugins/scaffolder-backend-module-github/src/actions/github.test.ts]: Removed, apologies, this got in from a bad rebase.

**Comment by benjdlambert** [comment]
@rewixe we need a changeset for the packages that have changed here in order to release them.

You can follow the instructions in the message here: https://github.com/backstage/backstage/pull/24501#issuecomment-2079452820

**Comment by rewixe** [comment]
@benjdlambert apologies, added changeset

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**